File: Java/1512. Number of Good Pairs.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
        int temp[] = new int[101], ans = 0;

        for(int i: nums) temp[i]++;
        for(int i: temp) ans += i * (i - 1) / 2;

        return ans;
    }
}

File: C++/1512. Number of Good Pairs.cpp
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        map<int, int> m;
        int res = 0;

        for(int i: nums) m[i]++;

        for(auto &[_, i]: m){
            res += i * ((int)i - 1) / 2;
        }

        return res;
    }
};
